src/OrleansProviders/Streams/Common/PooledCache/IObjectPool.cs

```diff
@@ -23,6 +23,31 @@ public interface IObjectPool<T>
         /// </summary>
         /// <param name="resource"></param>
         void Free(T resource);
+
+        /// <summary>
+        /// Try to find in-use object in this pool by Id 
+        /// </summary>
+        /// <param name="Id"></param>
+        /// <param name="inUseObject"></param>
+        /// <returns></returns>
+        bool TryFindInUseObject(object Id, out T inUseObject);
+    }
+
+    internal static class PurgeRequest
+    {
+        /// <summary>
+        /// Empty purge request which is just to inform cache with TimeBasedPurgePredicate to do 
+        /// a time based purge
+        /// </summary>
+        public static IDisposable EmptyPurgeRequest = EmptyDisposable.Instance.Value;
+
+        public class EmptyDisposable : IDisposable
+        {
+            public readonly static Lazy<EmptyDisposable> Instance = 
```

Comment:
what's the purpose of having a Lazy here, if you then have a static field that reads the value immediately?